### PR TITLE
Implement a basic type conversion support

### LINKED ISSUE #234: [2.x] Date fields aren't converted to object
When searching on tables, we have automatically conversions from database types to PHP equivalents, but on indexes, they seen always text type.

As Index can have Mapping definition, including text, boolean, numeric, timestamp, how can I teach Index / Document to convert these types to PHP equivalents?

### PR DESCRIPTION
**Pull request by CauanCabral**
Resolves #234

### CODE REVIEW

File: src/Datasource/Dummy.php [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ */
+namespace Cake\ElasticSearch\Datasource;
+
+use Cake\Database\Driver;
+
+/**
+ * Represents a dummy driver used only for convertions
+ * between types of ElasticSearch and PHP.
+ */
+class Dummy extends Driver
```
Comment (markstory) [inline src/Datasource/Dummy.php:25]: Why did you take this approach? The elastic search plugin doesn't use database drivers. Instead could we implement the DriverInterface?

Comment (CauanCabral) [reply src/Datasource/Dummy.php:25]: I tried to change as litle as possible the code base, but `Cake\Database\Type` system always expect a `Driver` instance on its methods.

As implementing a full `Driver` and separate `Connection` from him, I will need to rewrite many parts of current codebase.

Comment (markstory) [reply src/Datasource/Dummy.php:25]: Ah. We might need a driver instance in 3.x compatible code but the 4.x compatible code should be able to use an interface.

**Comment by CauanCabral** [comment]
Test included.

File: src/Datasource/Dummy.php (hunk 1 above)
Comment (othercorey) [reply src/Datasource/Dummy.php:25]: We should mark this `@internal` at least and probably name it `DummyDriver`.

File: src/Datasource/Dummy.php [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+{
+    /**
+     * Establishes a connection to the database server.
+     *
+     * @return bool True on success, false on failure.
+     */
+    public function connect()
```
Comment (othercorey) [inline src/Datasource/Dummy.php:32]: All of these can `@inheritDoc`

Comment (CauanCabral) [reply src/Datasource/Dummy.php:32]: I tried that, but codesniffer triggered error.

Should I ignore it (cs error)?

File: src/Datasource/Dummy.php (hunk 1 above)
Comment (CauanCabral) [reply src/Datasource/Dummy.php:25]: Current drivers doesn't include suffix _Driver_. Maybe follow namespace pattern?

`ElasticSearch\Database\Driver\Dummy.php` ?

File: src/Datasource/Dummy.php (hunk 2 above)
Comment (othercorey) [reply src/Datasource/Dummy.php:32]: I have no idea without seeing the error.

Comment (CauanCabral) [reply src/Datasource/Dummy.php]: My fault, I had put `{@inheritDoc}` instead of `@inheritDoc`. Fixed now.

File: src/Datasource/Dummy.php (hunk 1 above)
Comment (CauanCabral) [reply src/Datasource/Dummy.php]: Moved to `Database\Driver` namespace and put a `@internal` mark.

Also, removed `strict` as 2.x branch don't have it in any file.

**Pull request closed without merging**